include/dfu/mcuboot.h

```diff
@@ -8,11 +8,115 @@
 #ifndef __MCUBOOT_H__
 #define __MCUBOOT_H__
 
+#include <stdbool.h>
+#include <stddef.h>
+
+#include <zephyr/types.h>
+
+/**
+ * @brief MCUboot image header representation for image version
+ *
+ * The header for an MCUboot firmware image contains an embedded
+ * version number, in semantic versioning format. This structure
+ * represents the information it contains.
+ */
+struct mcuboot_img_sem_ver {
+	u8_t major;
+	u8_t minor;
+	u16_t revision;
+	u32_t build_num;
+};
+
+/**
+ * @brief Model for the MCUboot image header as of version 1
```

Comment:
Model = flash layout etc.

Reply:
What do you mean? Sorry, I don't understand.

This is not the contents of the image header as its laid out flash. It's a model for the relevant parts. See the `v1_raw` structure elsewhere in the patch for the actual contents on flash, or check the MCUboot design document.

Reply:
now understand - thought that it was  a model for data in flash (strange one as i didn't see appropriate structure in mcuboot). Model word confused me. But as you explain - ignore my previous comments.